fix(minifier): keep variables that are modified by `typeof foo === 'object' && foo !== null` compression

sapphi-red opened the pull request:
`typeof foo === 'object' && foo !== null` compression was not keeping the reference information and that caused used variables to be removed.
```js
var bar = foo.bar;
if (typeof bar !== 'object' || bar === null) console.log('foo')
```
was compressed into
```js
if (typeof foo.bar !== 'object' || bar === null) console.log('foo')
```

Boshen commented:
### Merge activity

* **Aug 25, 12:07 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

sapphi-red committed:
fix(minifier): keep variables that are modified by `typeof foo === 'object' && foo !== null` compression (#13290)

`typeof foo === 'object' && foo !== null` compression was not keeping the reference information and that caused used variables to be removed.
```js
var bar = foo.bar;
if (typeof bar !== 'object' || bar === null) console.log('foo')
```
was compressed into
```js
if (typeof foo.bar !== 'object' || bar === null) console.log('foo')
```
